packages/contracts-bedrock/src/L1/opcm/OPContractsManagerMigrator.sol

```diff
@@ -0,0 +1,279 @@
+        }
+    }
+
+    /// @notice Migrates a single portal to the new ETHLockbox and AnchorStateRegistry.
+    /// @param _systemConfig The system config for the chain being migrated.
+    /// @param _portalImpl The new OptimismPortalInterop implementation address.
+    /// @param _newLockbox The new ETHLockbox.
+    /// @param _newASR The new AnchorStateRegistry.
+    function _migratePortal(
+        ISystemConfig _systemConfig,
+        address _portalImpl,
+        IETHLockbox _newLockbox,
+        IAnchorStateRegistry _newASR
+    )
+        internal
+    {
+        // Convert portal to interop portal interface, and grab existing ETHLockbox, ASR, and DGF.
+        IOptimismPortalInterop portal = IOptimismPortalInterop(payable(_systemConfig.optimismPortal()));
+        IETHLockbox existingLockbox = IETHLockbox(payable(address(portal.ethLockbox())));
+        IAnchorStateRegistry existingASR = portal.anchorStateRegistry();
+        IDisputeGameFactory existingDGF = IDisputeGameFactory(payable(address(portal.disputeGameFactory())));
+
+        // Upgrade the portal to the interop implementation. Pass the EXISTING anchor state
+        // registry and eth lockbox to initialize so we don't change them yet.
+        _upgrade(
+            _systemConfig.proxyAdmin(),
+            address(portal),
+            _portalImpl,
+            abi.encodeCall(IOptimismPortalInterop.initialize, (_systemConfig, existingASR, existingLockbox))
+        );
+
+        // Authorize the portal on the new ETHLockbox.
+        _newLockbox.authorizePortal(IOptimismPortal(payable(address(portal))));
+
+        // Authorize the existing ETHLockbox to use the new ETHLockbox.
+        _newLockbox.authorizeLockbox(existingLockbox);
+
+        // Migrate the existing ETHLockbox to the new ETHLockbox.
+        existingLockbox.migrateLiquidity(_newLockbox);
+
+        // Clear out any implementations that might exist in the old DisputeGameFactory proxy.
+        // We clear out all potential game types to be safe.
+        existingDGF.setImplementation(GameTypes.CANNON, IDisputeGame(address(0)), hex"");
+        existingDGF.setImplementation(GameTypes.SUPER_CANNON, IDisputeGame(address(0)), hex"");
+        existingDGF.setImplementation(GameTypes.PERMISSIONED_CANNON, IDisputeGame(address(0)), hex"");
+        existingDGF.setImplementation(GameTypes.SUPER_PERMISSIONED_CANNON, IDisputeGame(address(0)), hex"");
+        existingDGF.setImplementation(GameTypes.CANNON_KONA, IDisputeGame(address(0)), hex"");
+        existingDGF.setImplementation(GameTypes.SUPER_CANNON_KONA, IDisputeGame(address(0)), hex"");
+
+        // Enable the ETH lockbox feature on the SystemConfig if not already enabled.
+        // This is needed for the SystemConfig's paused() function to use the correct identifier.
+        if (!_systemConfig.isFeatureEnabled(Features.ETH_LOCKBOX)) {
+            _systemConfig.setFeature(Features.ETH_LOCKBOX, true);
+        }
```

Comment:
OPCMV1's migrate functions doesn't do this AFAICT. Was that a bug?

Reply:
Sort of. It's actually not necessary to do this because the Interop portal just assumes the feature and doesn't check for it. We're going to need to reconcile all of this before we go into prod with interop but for now I'm just doing this prematurely. It works perfectly well without it.